python/pyspark/pandas/data_type_ops/boolean_ops.py

```diff
@@ -137,21 +136,13 @@ def mod(self, left: IndexOpsLike, right: Any) -> SeriesOrIndex:
             raise TypeError(
                 "Modulo can not be applied to %s and the given type." % self.pretty_name
             )
-        spark_session = left._internal.spark_frame.sparkSession
-
-        def safe_mod(left_col: PySparkColumn, right_val: Any) -> PySparkColumn:
-            if is_ansi_mode_enabled(spark_session):
```

Comment:
Just curious why don't we need to check ansi_mode here anymore?

Reply:
Bools are considered numeric and will take num_ops logic if that makes sense